setup.py

```diff
@@ -29,22 +29,40 @@
 "gfortran" and "python-dev" packages installed.
 '''
 
-import setuptools
-import os
-import sys
+import glob, os, setuptools, shutil, site, sys
 from os.path import join
 
 import nose
 from numpy.distutils.core import setup, Extension
+from setuptools.command.develop import develop
+from setuptools.command.install import install
 
+try:
+    from holopy import __version__
+except ImportError:
+    __version__ = 'unknown'
 HOLOPY_NOSE_PLUGIN_LOCATION = ('holopycatchwarnings = '
                                'holopy.core.tests.common:HoloPyCatchWarnings')
 
 hp_root = os.path.dirname(os.path.realpath(sys.argv[0]))
 
-# this will automatically build the scattering extensions, using the
-# setup.py files located in their subdirectories
-def configuration(parent_package='',top_path=' '):
+class PostDevelopConfig(develop):
+    """Post-installation for development mode."""
+    def run(self):
+        develop.run(self)
+        if os.name == 'nt': _move_S_msvc_libs('develop')
+
+
+class PostInstallConfig(install):
+    """Post-installation for installation mode."""
+    def run(self):
+        install.run(self)
+        if os.name == 'nt': _move_msvc_libs('install')
```

Comment:
Is there a way to combine these functions into one? They have a lot of repeated code...

Reply:
I took a look at the source of setuptools, and the `install` and `develop` classes really are quite different (not even the same base class) even though they have the same API. Maybe there's something clever that we can do like have some class devrive`develop` and `install`